src/sentry/replays/consumers/recording/process_recording.py

```diff
@@ -99,38 +110,45 @@ def _store(
         with sentry_sdk.start_transaction(
             op="replays.consumer.flush_batch", description="Replay recording segment stored."
         ):
-            sentry_sdk.set_extra("replay_id", message_dict["replay_id"])
-
-            try:
-                headers, recording_segment = self._process_headers(
-                    cached_replay_recording_segment.data
+            with segment_cache_manager(cached_replay_recording_segment) as cached_data:
+                sentry_sdk.set_extra("replay_id", message_dict["replay_id"])
+
+                try:
+                    headers, recording_segment = self._process_headers(cached_data)
+                except MissingRecordingSegmentHeaders:
+                    logger.warning(f"missing header on {message_dict['replay_id']}")
+                    return None
+
+                # Check for an existing recording-segment before continuing.  If a segment exists,
+                # cleanup and exit.
+                count_existing_segments = ReplayRecordingSegment.objects.filter(
+                    replay_id=message_dict["replay_id"],
+                    project_id=message_dict["project_id"],
+                    segment_id=headers["segment_id"],
+                ).count()
+                if count_existing_segments > 0:
+                    logger.warning("Recording segment was already processed.")
+                    return None
```

Comment:
Can we still capture this in sentry as an error so we have some visibility into it? I'm still working on improving this from the SDK side.